nadlogar/problems/models.py

```diff
@@ -239,3 +239,149 @@ def generate(self):
         razstavljen = sympy.factor(izraz)
 
         return {"izraz": sympy.latex(izraz), "razstavljen": sympy.latex(razstavljen)}
+
+
+class ElementiMnozice(Problem):
+    """Problem za izpis elementov množice iz podanega predpisa"""
+
+    linearna_kombinacija = models.BooleanField(
+        "linearna kombinacija",
+        help_text="Ali naj naloga vsebuje linearno kombinacijo?",
+        choices=[(True, "Da"), (False, "Ne")],
+    )
+
+    class Meta:
+        verbose_name = "elementi množice"
+
+    def generate(self):
+        pogoj = random.choice(["|", "<", "<="])
+        n = sympy.symbols("n")
+        if not self.linearna_kombinacija:
+            a = 1
+            b = 0
+        else:
+            a = random.randint(1, 3)
+            b = random.choice([-2, -1, 0, 1, 2])
+        if pogoj == "|":
+            stevilo = random.randint(15, 45)
+            ustrezni = sympy.divisors(stevilo)
+        elif pogoj == "<":
+            stevilo = random.randint(5, 12)
+            ustrezni = list(range(1, stevilo))
+        elif pogoj == "<=":
+            stevilo = random.randint(5, 8)
+            ustrezni = list(range(1, stevilo + 1))
+        mnozica = sympy.FiniteSet(*[a * x + b for x in ustrezni if a * x + b > 0])
+        pogoj_latex = {"|": r"\mid", "<": r"\lt", "<=": r"\le"}
+        return {
+            "n": sympy.latex(sympy.simplify(a * n + b)),
+            "pogoj": pogoj_latex[pogoj],
+            "stevilo": sympy.latex(stevilo),
+            "mnozica": sympy.latex(mnozica),
+        }
+
+
+class PotencnaMnozica(Problem):
+    """Problem za izpis potenčne množice od dane množice"""
+
+    class Meta:
+        verbose_name = "potenčna množica"
+
+    def generate(self):
+        velikost = random.randint(2, 3)
+        mnozice = [
+            ["a", "b", "c"],
+            [1, 2, 3],
+            ["x", "y", "z"],
+            ["alpha", "beta", "gamma"],
+            ["Pi", "Phi", "Xi"],
+            [3, 6, 9],
+            [3, 7, 42],
+        ]
+        mnozica = sympy.FiniteSet(*random.choice(mnozice)[:velikost])
+        potencna = mnozica.powerset()
+        return {"mnozica": sympy.latex(mnozica), "potencna": sympy.latex(potencna)}
+
+
+class OperacijeMnozic(Problem):
+    """Naloga za zapis unije, preseka, razlike, in kartezičnega produkta množic."""
+
+    class Meta:
+        verbose_name = "operacije z množicami"
+
+    @staticmethod
+    def generiraj_mnozico(velikost, od, do):
```

Comment:
Zdi se mi, da bo takih statičnih metod precej. Je kakšen boljši način za organizacijo kot ta, da jih "naključno" razdelimo med probleme?

Reply:
Ja, se strinjam ja. To bi po mojem spadalo v nek utility (idealno se tja prekane prvič ko rabiš dvakrat, pa refactora staro takrat, čeprav ni res realistično pričakovati da se bo to dogajalo...)